src/MakePotCommand.php

```diff
@@ -140,7 +153,7 @@ protected function retrieve_main_file_data() {
 
 		$files = new IteratorIterator( new DirectoryIterator( $this->source ) );
 
-		/** @var DirectoryIterator $file */
+		/* @var DirectoryIterator $file */
```

Comment:
This should be a real docblock `/** */`, but inside of the `foreach` block as the first line, as `$file` only exists within the `foreach` block.
This will then allow an IDE to know how to handle `$file` in there.

Reply:
@schlessera What IDE do you use? PhpStorm for example handles this just fine outside the `foreach`.

In WordPress core both `/**` and `/*` is used for these annotations. Not sure if WPCS would think the former is a badly formatted doc block.

Reply:
I use PHPStorm as well, and for a few versions now, it interprets this correctly. But technically, the variable does not exist within the scope where you qualify it, and some other editors and static analysis tools will flag this.

I usually do one of these two instead:
```
foreach ( $files as $file ) {
	/** @var DirectoryIterator $file */
	if ( $file->isFile() && $file->isReadable() && 'php' === $file->getExtension()) {
		$plugin_files[] = $file->getRealPath();
	}
}
```
```
foreach ( $files as /** @var DirectoryIterator $file */ $file ) {
	if ( $file->isFile() && $file->isReadable() && 'php' === $file->getExtension()) {
		$plugin_files[] = $file->getRealPath();
	}
}
```

Regarding the docblock:
Although PHPStorm is generally very lax in this regard, using `@var` inside of a regular multiline comment does not have any semantic meaning whatsoever. Tags only have a proper meaning within docblocks. Again, some tools are more strict about this and will ignore the `@var` if it is not within a proper docblock.

Reply:
Hmm, nevermind my comments about moving the docblock inside of the `foreach`. This is apparently handled as an exception in the proposed PSR: https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md

<img width="883" alt="image 2018-07-04 at 9 41 08 am" src="https://user-images.githubusercontent.com/83631/42263520-8da7e4f4-7f6e-11e8-8dba-8a2b102cb9ca.png">

Reply:
However, the above standard proposal also states that tags need to be within a docblock, not a regular comment.